tasks/service/async/src/main/java/org/projectnessie/nessie/tasks/async/testing/TestingTasksAsync.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.nessie.tasks.async.testing;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Supplier;
+import org.projectnessie.nessie.tasks.async.ScheduledHandle;
+import org.projectnessie.nessie.tasks.async.TasksAsync;
+
+/**
+ * {@link TasksAsync} implementation for testing purposes. Collects tasks to run in an ordered
+ * collection, tasks are run by calling {@link #doWork()}. There are no threads nor is there any
+ * other concurrent code involved. Maybe (not) surprisingly, this class is <em>not</em> thread safe.
+ */
+public class TestingTasksAsync implements TasksAsync {
```

Comment:
Could this be moved to a `testFixtures` folder?

Reply:
moved to test/src/main